include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -173,18 +175,35 @@ class geometry_to_segment_or_box
     >
     struct intersects
     {
-        static inline bool apply(Geometry const& g1, SegOrBox const& g2, Strategy const&)
+        typedef typename point_type<SegOrBox>::type point_type;
+
+        typedef segment_intersection_points<point_type> intersection_return_type;
+
+        static inline intersection_return_type
+        apply(Geometry const& g1, SegOrBox const& g2, Strategy const& s)
         {
-            return geometry::intersects(g1, g2);
+            return geometry::detail::disjoint::disjoint_with_info
+                       <Geometry, SegOrBox>::apply(
+                          g1, g2, s.get_relate_segment_segment_strategy());
```

Comment:
AFAIR the correct indentation is:
```
return geometry::detail::disjoint::disjoint_with_info
    <
        Geometry, SegOrBox
    >::apply(g1, g2, s.get_relate_segment_segment_strategy());
```

Reply:
or 
```
return geometry::detail::disjoint::disjoint_with_info
              <
                  Geometry,
                  SegOrBox
              >::apply(g1, g2, s.get_relate_segment_segment_strategy());
```
since typename parameters should be on different lines